packages/flutter_tools/lib/src/commands/run_mojo.dart

```diff
@@ -108,17 +107,30 @@ class RunMojoCommand extends FlutterCommand {
 
     if (argResults['android']) {
       args.add('--android');
-      final String appName = path.basename(appPath);
-      final String appDir = path.dirname(appPath);
-      args.add('mojo:launcher http://app/$appName');
-      args.add('--map-origin=http://app/=$appDir');
+    }
+
+    final Uri appUri = Uri.parse(bundlePath);
```

Comment:
I'd probably rename bundlePath to something that indicates that it might not be a path.  Maybe "targetApp" ?